### PR TITLE
fix(hotfix): cors not properly configured

### PR DESCRIPTION
**Pull request by arein**
# Description

Our CORS config doesn't work. This is a hotfix that makes it work.

## How Has This Been Tested?

Added functional tests to confirm the before state was broken and the after state works.

`cargo test`

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -150,12 +150,13 @@ pub async fn bootstap(mut shutdown: broadcast::Receiver<()>, config: Config) ->
             global_middleware.clone().layer(
                 CorsLayer::new()
                     .allow_methods([Method::GET, Method::POST, Method::DELETE])
-                    .allow_origin(
```
Comment (HarryET) [inline src/lib.rs:153]: Can you just update the terraform to * instead of a code change for that?

**Review by HarryET — APPROVED** [review]

**Review by xav — APPROVED** [review]
Can you fix the clippy warning?

File: src/lib.rs (hunk 1 above)
Comment (arein) [reply src/lib.rs:153]: That didn't work for me when I tried to test it in the functional tests.

Comment (arein) [reply src/lib.rs:153]: I think we'll need to do it slightly different.

**Pull request merged**